gems/sorbet-runtime/lib/types/types/typed_array.rb

```diff
@@ -13,7 +13,7 @@ def underlying_class
     end
 
     # @override Base
-    def valid?(obj)
+    def valid?(obj, deep=false)
       obj.is_a?(Array) && super
```

Comment:
ooc, we could short circuit all of these is `deep` is `true`, right? Why have you chosen to unconditionally call `super`?